tululbot/__init__.py

```diff
@@ -0,0 +1,96 @@
+from os import environ
+
+from dotenv import parse_dotenv
+from flask import Flask, request, jsonify
+
+from .utils import leli, QuoteEngine, TululBot
+
+
+def create_app(dotenv_path=None, config_dict=None):
+    """Return a Flask application object configured with the given argument.
+
+    At least one of `dotenv_path` or `config_dict` must be given. `dotenv_path` is
+    a path to where `.env` file is located. It will then be loaded with
+    `python-dotenv`'s `parse_dotenv` method. `config_dict` is a dictionary
+    of configuration values. If both are present, `dotenv_path` takes precedence.
+    """
+    if config_dict is None and dotenv_path is None:
+        raise ValueError('config_dict and dotenv_path cannot be both None')
+
+    app = Flask(__name__)
+    if config_dict is not None:
+        app.config.update(config_dict)
+    if dotenv_path is not None:
+        for k, v in parse_dotenv(dotenv_path):
+            app.config[k] = v
+
+    # Why do this? See https://core.telegram.org/bots/api#setwebhook
+    base_path = '/{}'.format(app.config['TELEGRAM_BOT_TOKEN'])
+
+    # Configure application logging
+    app.logger.setLevel(app.config['LOG_LEVEL'])
+
+    @app.route(base_path, methods=['POST'])
+    def main():
+        body = request.get_json()
+        app.logger.debug('update_id is %s', body['update_id'])
+        message = body.get('message')
+        app.logger.debug('message is %s', message)
+        if message is not None:
+            text = message.get('text')
+            app.logger.debug('text is %s', text)
+            chat_id = message['chat']['id']
+            app.logger.debug('chat_id is %s', chat_id)
+
+            if text is not None:
+                def reply(reply_text, disable_preview=False):
+                    payload = {
+                        'method': 'sendMessage',
+                        'chat_id': chat_id,
+                        'text': reply_text,
+                        'disable_web_page_preview':
+                            'true' if disable_preview else 'false',
+                        'reply_to_message_id': message['message_id']
+                    }
+                    return jsonify(**payload)
+
+                bot = TululBot()
+
+                @bot.command(r'^/leli (?P<term>.+)$')
```

Comment:
I think we should decouple this. Initialization, data sanitation, and execution still happen in one big block.